boa_engine/src/syntax/ast/expression/access.rs

```diff
@@ -0,0 +1,232 @@
+use crate::syntax::ast::{expression::Expression, ContainsSymbol};
+use boa_interner::{Interner, Sym, ToInternedString};
+
+#[cfg_attr(feature = "deser", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Debug, PartialEq)]
+pub enum PropertyAccessField {
+    Const(Sym),
+    Expr(Box<Expression>),
```

Comment:
Do we need these to be boxed? I guess it's a space optimization, right?

Reply:
Yeah, we need to box it. `Expression` contains `PropertyAccess`, which contains `PropertyAccessField`, meaning we can't stack-allocate `PropertyAccessField` without boxing either the whole `PropertyAccessField` or `Expression`, and I think it's a lot more space and speed efficient to box the `Expression` and leave `Sym` on the stack, which is pointer-sized.